src/hb-ot-map.cc

```diff
@@ -390,5 +390,28 @@ hb_ot_map_builder_t::compile (hb_ot_map_t                  &m,
   }
 }
 
+unsigned int hb_ot_map_t::get_features (unsigned int start_offset, unsigned int *feature_count, hb_feature_t *out_features) const
+{
+  if (feature_count)
+  {
+    auto sub_features = features.as_array ().sub_array (start_offset, *feature_count);
+    *feature_count = sub_features.length;
```

Comment:
If you pass the pointer to `sub_array` instead of dereferencing it, it sets the `feature_count` for you automagically.